test/model/internal_link_test.dart

```diff
@@ -0,0 +1,394 @@
+
+import 'package:checks/checks.dart';
+import 'package:test/scaffolding.dart';
+import 'package:zulip/api/model/model.dart';
+import 'package:zulip/model/internal_link.dart';
+import 'package:zulip/model/narrow.dart';
+import 'package:zulip/model/store.dart';
+
+import '../example_data.dart' as eg;
+import 'test_store.dart';
+
+PerAccountStore setupStore({
+  required Uri realmUrl,
+  List<ZulipStream>? streams,
+  List<User>? users,
+}) {
+  final account = eg.selfAccount.copyWith(realmUrl: realmUrl);
+  final store = eg.store(account: account);
+  if (streams != null) {
+    store.addStreams(streams);
+  }
+  store.addUser(eg.selfUser);
+  if (users != null) {
+    store.addUsers(users);
+  }
+  return store;
+}
+
+void main() {
+  final realmUrl = Uri.parse('https://example.com/');
+
+  checkExpectedNarrows(List<(String, Narrow?)> testCases, {
```

Comment:
nit: since this now contains a `test` call, its name should say "test" rather than "check" — that convention helps when reading callers, for seeing the nesting of `test` calls

(I see the corresponding test file in zulip-mobile doesn't get that right, oops.)